### PR TITLE
upload config file with yaml, not viper

### PR DESCRIPTION
**Pull request by davidbloss**
## Changelog

Update `kubectl config` to load its config with `yaml. Unmarshal()` rather than `viper.Unmarshal()`

- [X] List your changes here
- [X] Make a `changie` entry

### CODE REVIEW

File: src/cmd/root.go [hunk 1]
```diff
@@ -60,37 +60,13 @@ func init() {
...
 	cobra.OnInitialize(func() {
-		readConfig()
 		setupLogging()
 		setupOutput()
 		setupConcurrency()
 		setupAPIToken()
 	})
 }
 
-func readConfig() {
```
Comment (taimoor-at-opslevel) [inline src/cmd/root.go:71]: I think this function and the call to it on line 63 should be retained to a degree. We still need to do the operations on line 89, 90 afaict.

File: src/cmd/config.go [hunk 2]
```diff
@@ -70,16 +71,19 @@ func init() {
 }
 
 func LoadConfig() (*common.Config, error) {
-	v := &common.ConfigVersion{}
```
Comment (taimoor-at-opslevel) [inline src/cmd/config.go:73]: The `type ConfigVersion struct` in config.go should also be deleted since it's unused now.

File: src/cmd/root.go [hunk 3]
```diff
@@ -60,37 +60,13 @@ func init() {
...
-	if cfgFile != "" {
-		if cfgFile == "." {
```
Comment (taimoor-at-opslevel) [inline src/cmd/root.go:73]: Also, IIRC you can never parse `.` as an argument in the CLI, so this should have actually been `-`.

File: src/cmd/config.go [hunk 4]
```diff
@@ -70,16 +71,19 @@ func init() {
...
-	viper.Unmarshal(&v)
-	if v.Version != common.ConfigCurrentVersion {
-		return nil, fmt.Errorf("supported config version is '%s' but found '%s' | Please update config file or create a new sample with `kubectl opslevel config sample`", common.ConfigCurrentVersion, v.Version)
+	var commonConfig common.Config
+
+	yamlData, err := os.ReadFile(cfgFile)
```
Comment (taimoor-at-opslevel) [inline src/cmd/config.go:76]: Need to port over the logic from root.go `readConfig()` to:

1. If `cfgFile == "-"` read the config from stdin as yaml
2. If the file at location `cfgFile` is found, read it. This is how the code works now.
3. If the file at location `cfgFile` is not found, read the file `$HOME/opslevel.yaml`

**Review by taimoor-at-opslevel — CHANGES_REQUESTED** [review]

**Comment by taimoorgit** [comment]
See https://github.com/OpsLevel/kubectl-opslevel/pull/222

**Pull request closed without merging**